src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -368,12 +367,33 @@ protected virtual SamlConditions CreateConditions(SecurityTokenDescriptor tokenD
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(new Uri(tokenDescriptor.Audience)));
+            var uriList = CreateUriList(tokenDescriptor);
+            if (!uriList.IsNullOrEmpty())
+                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(uriList));
 
             return conditions;
         }
 
+        private static List<Uri> CreateUriList(SecurityTokenDescriptor tokenDescriptor)
+        {
+            var uriList = new List<Uri>();
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
+            {
+                foreach (var audience in tokenDescriptor.Audiences)
+                {
+                    if (audience != null)
+                    uriList.Add(new Uri(audience, UriKind.Absolute));
```

Comment:
This is a change in behavior.
Previously, the code did not have the UriKind.Absolute property, please revert here and elsewhere.

Reply:
@brentschmaltz Are you recommending we remove UriKind entirely or use something other than absolute? @keegan-caruso recommended using UriKind in [their comment here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2575#discussion_r1622846877) but perhaps I misunderstood which kind to use?

Reply:
@keegan-caruso  can you help bring clarity to this one? Thank you.

Reply:
It shouldn't be a change in behavior, the default kind is Absolute.

It is a bit unrelated but I thought it clarified the code a little, let's address it offline and revert for now. We can decide if we want to make the change later.

Reply:
removed